tests/pytorch/test_pytorch_autolog.py

```diff
@@ -457,3 +461,481 @@ def test_autolog_registering_model():
 
         registered_model = MlflowClient().get_registered_model(registered_model_name)
         assert registered_model.name == registered_model_name
+
+
+@pytest.mark.skipif(
+    Version(pl.__version__) < Version("1.6.0"),
+    reason="`Automatic model checkpointing doesn't exist in pytorch-lightning < 1.6.0",
+)
+def test_model_checkpoint_callback():
+    class FakeTrainer:
+        def __init__(self):
+            self.callback_metrics = {}
+            self.current_epoch = 0
+            self.global_step = 0
+
+    with mlflow.start_run() as run, mock.patch(
+        "mlflow.client.MlflowClient.log_dict"
```

Comment:
why need to mock this?

Reply:
in this file we dont't generate real checkpoint file so we have to mock mlflow `log_artifact` , but `log_dict` is implemented by calling `log_artifact`, so we have to mock it too.

Reply:
why not log artifacts and check they are logged?

Reply:
> why not log artifacts and check they are logged?

I have added another 2 integration tests `test_per_epoch_automatic_model_checkpoint` and `test_per_n_steps_automatic_model_checkpoint` without mocking.

In these mocking tests, I want to test all kinds of param combination, so mocking makes these tests lightweight and faster

Reply:
Can we use a small model and training dataset? If the model and training dataset are small, it should be quick to log checkpoints, right?

Reply:
Updated @harupy 
mocks are removed.